doc/source/cluster/kubernetes/examples/rayjob-kueue-gang-scheduling.md

```diff
@@ -0,0 +1,249 @@
+(kuberay-kueue-gang-scheduling-example)=
+
+# Gang Scheduling with RayJob and Kueue
+
+This guide demonstrates how to use Kueue for gang scheduling RayJob resources, taking advantage of dynamic resource provisioning and queueing on Kubernetes. The [Fine-tune a PyTorch Lightning Text Classifier with Ray Data](https://docs.ray.io/en/master/train/examples/lightning/lightning_cola_advanced.html) example will be used to illustrate the concepts.
+
+## What's Gang Scheduling?
+
+Gang scheduling in Kubernetes ensures that a group of related Pods (such as those in a Ray cluster)
+only start when all required resources are available. This is crucial when working with expensive,
+limited resources like GPUs.
+
+## What's Kueue?
+
+[Kueue](https://kueue.sigs.k8s.io/) is a Kubernetes-native system that manages quotas
+and how jobs consume them. Kueue decides when:
+* To make a job wait
+* To admit a job to start, meaning that Kubernetes creates Pods.
+* To preempt a job, meaning that Kubernetes deletes active Pods.
+
+Kueue has native support for some KubeRay APIs. Specifically, you can use Kueue
+to manage resources consumed by RayJob and RayCluster. See the
+[Kueue documentation](https://kueue.sigs.k8s.io/docs/overview/) to learn more.
+
+## Why use gang scheduling?
+
+Gang scheduling is essential when working with expensive, limited hardware accelerators like GPUs.
+It prevents partially provisioned Ray clusters that claim unused GPUs. With Kueue and gang scheduling,
+a RayJob will remain suspended until the Kubernetes cluster and the underlying cloud provider can
+guarantee the capacity needed to execute the RayJob. This greatly improves GPU utilization and cost,
+especially when GPU availability is limited.
+
+## Create a Kubernetes cluster on GKE
+
+Create a GKE cluster with autoscaling enabled:
+```bash
+gcloud container clusters create kuberay-gpu-cluster \
+    --num-nodes=1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+    --zone=us-west1-b --machine-type e2-standard-4 --cluster-version 1.29
+```
+
+Create a GPU node pool with the "enable-queued-provisioning" option enabled:
+```bash
+gcloud beta container node-pools create gpu-node-pool \
+  --accelerator type=nvidia-l4,count=1,gpu-driver-version=latest \
+  --enable-queued-provisioning \
+  --reservation-affinity=none  \
+  --zone us-west1-b \
+  --cluster kuberay-gpu-cluster \
+  --num-nodes 0 \
+  --min-nodes 0 \
+  --max-nodes 10 \
+  --enable-autoscaling \
+  --machine-type g2-standard-4
+```
+
+This command creates a node pool which initially has zero nodes. The `gcloud beta` command is used since some flags are still in Beta status.
```

Comment:
```suggestion
This command creates a node pool which initially has zero nodes. Use the `gcloud beta` command because some flags have beta status.
```